AppHandling/Run-AlPipeline.ps1

```diff
@@ -1746,31 +1784,58 @@ Write-Host -ForegroundColor Yellow @'
         }
     }
 
+    $appType = switch ($true) {
+        $app { "app" }
+        $testApp { "testApp" }
+        $bcptTestApp { "bcptApp" }
+        Default { "app" }
+    }
+
+    $compilationParams = $Parameters + $CopParameters
+
+    # Run pre-compile script if specified
+    if($PreCompileApp) {
+        Write-Host "Running custom pre-compilation script..."
+
+        Invoke-Command -ScriptBlock $PreCompileApp -ArgumentList $appType, ([ref] $compilationParams)
+    }
+
     try {
-        Write-Host "`nCompiling $($Parameters.appProjectFolder)"
-        $withCops = $Parameters+$CopParameters
+        Write-Host "`nCompiling $($compilationParams.appProjectFolder)"
         if ($useCompilerFolder) {
-            $appFile = Compile-AppWithBcCompilerFolder @parameters
+            $appFile = Compile-AppWithBcCompilerFolder @compilationParams
         }
         else {
-            $appFile = Invoke-Command -ScriptBlock $CompileAppInBcContainer -ArgumentList $withCops
+            $appFile = Invoke-Command -ScriptBlock $CompileAppInBcContainer -ArgumentList $compilationParams
         }
     }
     catch {
         if ($escapeFromCops) {
             Write-Host "Retrying without Cops"
+
+            # Remove Cops parameters
+            $compilationParamsCopy = $compilationParams.Clone()
+            $compilationParamsCopy.Keys | Where-Object {$_ -in $CopParameters.Keys} | ForEach-Object { $compilationParams.Remove($_) }
+
             if ($useCompilerFolder) {
-                $appFile = Compile-AppWithBcCompilerFolder @parameters
+                $appFile = Compile-AppWithBcCompilerFolder @compilationParams
             }
             else {
-                $appFile = Invoke-Command -ScriptBlock $CompileAppInBcContainer -ArgumentList $Parameters
+                $appFile = Invoke-Command -ScriptBlock $CompileAppInBcContainer -ArgumentList $compilationParams
             }
         }
         else {
             throw $_
         }
     }
 
+    # Run post-compile script if specified
+    if($PostCompileApp) {
+        Write-Host "Running custom post-compilation script..."
+
+        Invoke-Command -ScriptBlock $PostCompileApp -ArgumentList $appFile, $appType, $compilationParams

```

Comment:
Now this one contains a different parameter set whether or not the cops where removed
Is this on purpose?

Reply:
Well, yeah. 

The idea was the post compile script to get the _actual_ compilation parameters that were used to create the app file.